test/functional/tests/io_class/test_io_class_stats_file_size_core.py

```diff
@@ -0,0 +1,201 @@
+                        check_statistics(core, io_class, fio_stats, io_class_stats)
+                        fs_utils.remove(f"{core.mount_point}/*", force=True, recursive=True)
+                    size_min = size + Unit.Byte
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_io_class_stats_file_size_core_direct():
+    """
+    title: Open CAS statistics values for file size - per core.
+    description: Check Open CAS ability to assign correct IO class to file size.
+    pass_criteria:
+        - after FIO with direct statistics increases for direct IO class
+    """
+
+    with TestRun.step("Prepare devices."):
+        cache_devices, core_devices = prepare_devices()
+
+    with TestRun.step("Start caches (one for each supported cache mode) and add core devices."):
+        caches, cores = prepare_caches_and_cores(cache_devices, core_devices)
+
+    with TestRun.step("Make filesystem on OpenCAS devices and mount it."):
+        for core, fs in zip(cores, cycle(Filesystem)):
+            mount_point = core.path.replace('/dev/', '/mnt/')
+            core.create_filesystem(fs)
+            core.mount(mount_point)
+            sync()
+
+    with TestRun.step("In default IO class configuration file set Eviction priority=1 "
+                      "and load it for all caches."):
+        io_classes = prepare_io_classes(caches)
+
+    with TestRun.step("Run fio for all devices checking IO class statistics."):
+        sizes = get_sizes(io_classes)
+        size_min = Size(512, Unit.Byte)
+
+        io_class_direct = [io_class for io_class in io_classes if "direct" in io_class.rule][0]
+
+        for io_class, size in zip(io_classes[2:-1], sizes):
+            with TestRun.step(f"Testing IO class {io_class.id}."):
+                with TestRun.step("Run fio for each device with direct=1 and check that "
+                                  "for direct IO class statistics increases."):
+                    for core in cores:
+                        core.reset_counters()
+                        fio = fio_params(core, size_min, size, direct=True)
+                        result = fio.run()
+                        sync()
+                        drop_caches()
+                        fio_stats = \
+                            result[0].write_requests_number() + result[0].read_requests_number()
+                        io_class_stats = core.get_io_class_statistics(io_class_direct.id).\
+                            request_stats.requests_total
+                        check_statistics(core, io_class_direct, fio_stats, io_class_stats)
+                        fs_utils.remove(f"{core.mount_point}/*", force=True, recursive=True)
+                    size_min = size + Unit.Byte
+
+
+def fio_params(core, size_min, size_max, direct=False):
+    fio = Fio().create_command() \
+        .io_engine(IoEngine.libaio) \
+        .do_verify() \
```

Comment:
The tests focus on verifying the number of requests issued and handled. I think you can remove `do_verify()` and `.verification_with_pattern()`